### PR TITLE
Make aimec

### PR DESCRIPTION
**Pull request by matthiasto**
This is the AIMEC analysis module.
I also reorganized the different functions: for example, a lot of functions from com2AB were moved to in2Trans or in3Utils

### CODE REVIEW

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 1]
```diff
@@ -0,0 +1,1149 @@
...
+    depthfileList = [str(pathFlowHeight) +
+                     '/' +
+                     str(name) for name in
+                     sorted(os.listdir(pathFlowHeight)) if os.path.isfile(os.path.join(pathFlowHeight, name))]
+    cfgPath['depthfileList'] = depthfileList
+
+    massfileList = [str(pathMassBalance) +
+                    '/' +
+                    str(name) for name in
+                    sorted(os.listdir(pathMassBalance)) if os.path.isfile(os.path.join(pathMassBalance, name))]
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:70]: Maybe split this for readability? Or break out into function, as it does the same thing 3 times?

And use os.path.join wit os.pathsep  (to make it platform independent....)

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 2]
```diff
@@ -0,0 +1,1149 @@
...
+    defaultName = str(avalancheDir).split('/')[-1]
+    cfgPath['defaultName'] = defaultName
+
+    set_name = pressurefileList[0].split('/')[-4]
+    cfgPath['set_name'] = set_name
+    project_name = str(profileLayer).split('/')[-4]
+    cfgPath['project_name'] = project_name
+    path_name = str(profileLayer[0]).split('/')[-1]
+    cfgPath['path_name'] = path_name
+
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:85]: use the os.path functions like basename, path etc to get things like this..

File: avaframe/runAna3AIMEC.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+log.info('Comparing data to reference')
+doku, runout_doku, delta_h, elevRel = ana3AIMEC.analyzeDocu(pressureLimit,
+                                                            cfgPath['pressurefileList'],
+                                                            deskewedRasterInd,
+                                                            deskewedRasterPressure,
+                                                            deskewedRasterDepth,
+                                                            deskewedRasterPressure[0],
+                                                            deskewedRasterDEM,
+                                                            with_doku=False)
+
```
Comment (fso42) [inline avaframe/runAna3AIMEC.py:90]: Work with dicts more

File: avaframe/in2Trans/geoTrans.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+    ind = np.argmin(Dist)
+    indSplit = int(IndSplit[ind])
+    projSplitPoint = {}
+    projSplitPoint['x'] = AvaProfile['x'][indSplit]
+    projSplitPoint['y'] = AvaProfile['y'][indSplit]
+    projSplitPoint['z'] = AvaProfile['z'][indSplit]
+    projSplitPoint['s'] = AvaProfile['s'][indSplit]
+    projSplitPoint['indSplit'] = indSplit
+    return projSplitPoint
```
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/in2Trans/geoTrans.py:119]: Yes and no, if you change the variable names, they are very generic. Replacing splitPoint by point and find splitPoint by projectPointOnLine for example. We might often need to resample a line no? not only for AB?

[the remark this answers is not in the document]
Comment (fso42) [reply avaframe/in2Trans/geoTrans.py:119]: Good point, then rename it. 
And I suggest then changing the argument list from:
`def prepareLine(dem, avapath, splitPoint, distance):`
to
`def prepareLine(dem, avapath, distance, splitPoint=None):`
to make the splitPoint optional...

[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/in2Trans/geoTrans.py:119]: makes sense!

**Pull request merged**